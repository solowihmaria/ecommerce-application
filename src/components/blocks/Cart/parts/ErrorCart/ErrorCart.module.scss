@use '../../../../../assets/styles/constants.module' as c;

.errorCartContent {
    display: flex;
    flex-direction: column;
    gap: 5rem;
    align-items: center;
    justify-content: center;
    height: min-content;
    padding: 3rem 2rem;
    border: 1px solid c.$border;
    border-radius: 1.2rem;
    backdrop-filter: blur(15px);
}

.cartTitle {
    padding: 0 2rem;
    font-size: 2rem;
}

.errorCartText {
    color: c.$color-text-primary;
}

.button.errorCartButton {
    width: 100%;
    background-color: c.$color-accent-primary;
}

.errorCartButtonContainer {
    width: 100%;
    max-width: 32rem;
}
