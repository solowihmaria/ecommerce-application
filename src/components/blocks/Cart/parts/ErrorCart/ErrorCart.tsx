import { Link } from 'react-router-dom';
import { Button } from '../../../../ui/Button';
import styles from './ErrorCart.module.scss';
import { Heading } from '../../../../ui/Heading';
import clsx from 'clsx';

export const ErrorCart = ({ message }: { message: string }) => {
    return (
        <div
            data-testid="error-cart-test-id"
            className={styles.errorCartContent}
        >
            <Heading className={styles.cartTitle}>Shopping Cart</Heading>
            <p className={styles.errorCartText}>{message}</p>
            <Link to="/main" className={styles.errorCartButtonContainer}>
                <Button className={clsx(styles.errorCartButton, styles.button)}>
                    Go to Home
                </Button>
            </Link>
        </div>
    );
};
