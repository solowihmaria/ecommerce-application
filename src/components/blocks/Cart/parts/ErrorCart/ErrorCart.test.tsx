import '@testing-library/jest-dom';

import { ErrorCart } from './ErrorCart';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

describe('<ErrorCart/> component', () => {
    test('Render ErrorCart', () => {
        const message = 'Test error';
        render(
            <MemoryRouter>
                <ErrorCart message={message} />
            </MemoryRouter>
        );

        const errorCartContainer = screen.getByTestId('error-cart-test-id');
        expect(errorCartContainer).toBeInTheDocument();
        expect(errorCartContainer).toMatchSnapshot();
    });
});
