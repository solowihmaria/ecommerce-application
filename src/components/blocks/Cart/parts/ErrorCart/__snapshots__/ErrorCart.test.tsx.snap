// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorCart/> component Render ErrorCart 1`] = `
<div
  class="errorCartContent"
  data-testid="error-cart-test-id"
>
  <h1
    class="heading heading--h1 cartTitle"
  >
    Shopping Cart
  </h1>
  <p
    class="errorCartText"
  >
    Test error
  </p>
  <a
    class="errorCartButtonContainer"
    data-discover="true"
    href="/main"
  >
    <button
      class="button primary errorCartButton button"
    >
      Go to Home
    </button>
  </a>
</div>
`;
